src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -560,7 +679,47 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
                 Expression.Call(LogParameterBindingFailureMethod,
                     HttpContextExpr, parameterTypeNameConstant, parameterNameConstant, TempSourceStringExpr));
 
-            var tryParseCall = Expression.Call(tryParseMethod, TempSourceStringExpr, parsedValue);
+            // Before call the TryParse Method, we should know the exact param to inject
+            Type type = Nullable.GetUnderlyingType(parameter.ParameterType) ?? parameter.ParameterType;
+            var useNumberStyles = UseTryParseWithNumberStyleOption(type);
+            var useDateTimeStyles = UseTryParseWithDateTimeStyleOptions(type);
```

Comment:
This is feeling like a lot of repeated logic. I'm thinking that the `TryParseMethodCache` (now just `Cache` should change from a `ConcurrentDictionary<Type, MethodInfo?>` to `ConcurrentDictionary<Type, Func<Expression, MethodCallExpression>?>`, where the Func is the `CreateTryParseCall` mentioned above, but with the `MethodInfo tryParseMethod` captured since that never changes for a given `Type`.

Reply:
I got the point. I'll try a refacto 🙂